### DISCUSSION TITLE
Jacobian calculation with respect to two variables

### DISCUSSION THREAD

**Discussion by AmrIbrahim8894** [idaholab/moose#27557] [Q&A General]
Hi there,
I am developing an action for cohesive zone in poroelastodynamic media, I have a question regarding the jacobian calculation for the traction jump. If the traction jump is function of displacement jump only and if, i am understanding correctly, we use `computeQpJacobian` to calculate the diagonal entries of the jacobian matrix and `computeQpOffDiagJacobian` to calculate the Off-diagonal entries of the jacobian matrix. 
[
![2](https://github.com/idaholab/moose/assets/147007633/8e958d00-cf94-414a-802b-a03c49a6b1bd)
](url)
But, if the traction jump is function of both displacement jump and fluid velocity jump (another variable), the jacobian matrix will be as  follows: 
![1](https://github.com/idaholab/moose/assets/147007633/77f1ca8c-a165-4513-8f22-8cdde4afc725)
My question is how to implement this new jacobian matrix in moose?
Thank you.

**Comment by lindsayad** [comment]
You still use the `computeQpOffDiagJacobian` method. You just write different Jacobian statements depending on the passed in `jvar`. Alternatively you can use automatic differentiation and not worry about writing the Jacobians
